sw/device/silicon_creator/lib/drivers/otp_unittest.cc

```diff
@@ -27,47 +29,55 @@ class OtpReadTest : public OtpTest {
 };
 
 TEST_F(OtpReadTest, Read32) {
-  EXPECT_READ32(offset_, 0x00010203);
+  EXPECT_SEC_READ32(mmio_, base_ + offset_, 0x00010203);
 
-  EXPECT_EQ(otp_read32(&otp_, 0), 0x00010203);
+  EXPECT_EQ(otp_read32(0), 0x00010203);
 }
 
 TEST_F(OtpReadTest, Read64) {
-  EXPECT_READ32(offset_ + 8, 0x04050607);
-  EXPECT_READ32(offset_ + 4, 0x08090A0B);
+  EXPECT_SEC_READ32(mmio_, base_ + offset_ + 8, 0x04050607);
+  EXPECT_SEC_READ32(mmio_, base_ + offset_ + 4, 0x08090A0B);
 
-  EXPECT_EQ(otp_read64(&otp_, 4), 0x0405060708090A0B);
+  EXPECT_EQ(otp_read64(4), 0x0405060708090A0B);
 }
 
 TEST_F(OtpReadTest, ReadLen32) {
-  EXPECT_READ32(offset_, 0x08090A0B);
+  EXPECT_SEC_READ32(mmio_, base_ + offset_, 0x08090A0B);
 
   uint32_t value = 0;
-  otp_read(&otp_, 0, &value, sizeof(value));
+  EXPECT_EQ(otp_read(0, &value, sizeof(value)), kErrorOk);
   EXPECT_EQ(value, 0x08090A0B);
 }
 
 TEST_F(OtpReadTest, ReadLen64) {
-  EXPECT_READ32(offset_, 0x0C0D0E0F);
-  EXPECT_READ32(offset_ + 4, 0x08090A0B);
+  EXPECT_SEC_READ32(mmio_, base_ + offset_, 0x0C0D0E0F);
+  EXPECT_SEC_READ32(mmio_, base_ + offset_ + 4, 0x08090A0B);
 
   uint64_t value = 0;
-  otp_read(&otp_, 0, &value, sizeof(value));
+  EXPECT_EQ(otp_read(0, &value, sizeof(value)), kErrorOk);
   EXPECT_EQ(value, 0x08090A0B0C0D0E0F);
 }
 
 TEST_F(OtpReadTest, ReadLenN) {
   for (int val = 0; val < 16; ++val) {
-    EXPECT_READ32(offset_ + val * sizeof(uint32_t), val);
+    EXPECT_SEC_READ32(mmio_, base_ + offset_ + val * sizeof(uint32_t), val);
   }
 
   std::vector<uint32_t> arr(16);
-  otp_read(&otp_, 0, &arr[0], arr.size() * sizeof(uint32_t));
+  EXPECT_EQ(otp_read(0, &arr[0], arr.size() * sizeof(uint32_t)), kErrorOk);
 
   for (int i = 0; i < arr.size(); ++i) {
     EXPECT_EQ(arr[i], i);
```

Comment:
This would be better with a specialized matcher:

```
EXEPCT_THAT(arr, ElementsAre(0, 1, 2, <etc>));
```

or by initializing an `expected_values` array and then `EXPECT_THAT(arr, ElementsAreArray(expected_values))`.

Reply:
I went with your second suggestion. Is there any reason for the `ElementsAreArray` wrapper here? I didn't include it as it appears to just add unnecessary verbosity to the failure case.

Reply:
The extra verbosity is the point :)

I'm somewhat surprised that `EXPECT_THAT` seems to perform the correct match without the matcher (well, not _too_ suprised).  The matcher would be more useful if you were checking a more complex type or had a more complex pattern of data.  Ie: somethine like `ElementsAre(Eq(0), Gt(1), Lt(10))` or `EqualsProto`/`EquivToProto`.

Reply:
Oh right, this makes total sense. Done.